### Step 1. 필요한 라이브러리 임포트

In [1]:
import pandas as pd

### Step 2. [데이터](https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user) 확인

- user_id 유저 아이디
- age 나이
- gender 성별
- occupation 직업
- zip_code 우편번호

### Step 3. 

**read_csv를 통해 데이터를 올바른 형태로 불러온 뒤 users에 저장해주세요.**

user_id를 인덱스로 설정해주세요.

In [2]:
users = pd.read_csv('https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user', sep='|')
users.head()

user_id  age gender  occupation zip_code
0        1   24      M  technician    85711
1        2   53      F       other    94043
2        3   23      M      writer    32067
3        4   24      M  technician    43537
4        5   33      F       other    15213

In [5]:
users1 = users.set_index('user_id')
users1.head()

age gender  occupation zip_code
user_id                                 
1         24      M  technician    85711
2         53      F       other    94043
3         23      M      writer    32067
4         24      M  technician    43537
5         33      F       other    15213

### Step 4. 직업별 평균 연령을 구해주세요

occupation
administrator    38.746835
artist           31.392857
doctor           43.571429
educator         42.010526
engineer         36.388060
entertainment    29.222222
executive        38.718750
healthcare       41.562500
homemaker        32.571429
lawyer           36.750000
librarian        40.000000
marketing        37.615385
none             26.555556
other            34.523810
programmer       33.121212
retired          63.071429
salesman         35.666667
scientist        35.548387
student          22.081633
technician       33.148148
writer           36.311111
Name: age, dtype: float64

In [7]:
users1.groupby('occupation')['age'].mean()

occupation
administrator    38.746835
artist           31.392857
doctor           43.571429
educator         42.010526
engineer         36.388060
entertainment    29.222222
executive        38.718750
healthcare       41.562500
homemaker        32.571429
lawyer           36.750000
librarian        40.000000
marketing        37.615385
none             26.555556
other            34.523810
programmer       33.121212
retired          63.071429
salesman         35.666667
scientist        35.548387
student          22.081633
technician       33.148148
writer           36.311111
Name: age, dtype: float64

### Step 5. 직업별 남성 비율을 구한 뒤 내림차순으로 정렬해주세요

doctor           100.000000
engineer          97.014925
technician        96.296296
retired           92.857143
programmer        90.909091
executive         90.625000
scientist         90.322581
entertainment     88.888889
lawyer            83.333333
salesman          75.000000
educator          72.631579
student           69.387755
other             65.714286
marketing         61.538462
writer            57.777778
none              55.555556
administrator     54.430380
artist            53.571429
librarian         43.137255
healthcare        31.250000
homemaker         14.285714
dtype: float64

In [71]:
users1.head()

age gender  occupation zip_code
user_id                                 
1         24      M  technician    85711
2         53      F       other    94043
3         23      M      writer    32067
4         24      M  technician    43537
5         33      F       other    15213

In [19]:
tmp = users1.groupby(['occupation', 'gender']).count()
tmp

age  zip_code
occupation    gender               
administrator F        36        36
              M        43        43
artist        F        13        13
              M        15        15
doctor        M         7         7
educator      F        26        26
              M        69        69
engineer      F         2         2
              M        65        65
entertainment F         2         2
              M        16        16
executive     F         3         3
              M        29        29
healthcare    F        11        11
              M         5         5
homemaker     F         6         6
              M         1         1
lawyer        F         2         2
              M        10        10
librarian     F        29        29
              M        22        22
marketing     F        10        10
              M        16        16
none          F         4         4
              M         5         5
other         F        36        36
              M        69        69
programmer    F         6         6
              M        60        60
retired       F         1         1
              M        13        13
salesman      F         3         3
              M         9         9
scientist     F         3         3
              M        28        28
student       F        60        60
              M       136       136
technician    F         1         1
              M        26        26
writer        F        19        19
              M        26        26

In [52]:
#tmp.loc[(:,'M')]
tmp.index

MultiIndex([('administrator', 'F'),
            ('administrator', 'M'),
            (       'artist', 'F'),
            (       'artist', 'M'),
            (       'doctor', 'M'),
            (     'educator', 'F'),
            (     'educator', 'M'),
            (     'engineer', 'F'),
            (     'engineer', 'M'),
            ('entertainment', 'F'),
            ('entertainment', 'M'),
            (    'executive', 'F'),
            (    'executive', 'M'),
            (   'healthcare', 'F'),
            (   'healthcare', 'M'),
            (    'homemaker', 'F'),
            (    'homemaker', 'M'),
            (       'lawyer', 'F'),
            (       'lawyer', 'M'),
            (    'librarian', 'F'),
            (    'librarian', 'M'),
            (    'marketing', 'F'),
            (    'marketing', 'M'),
            (         'none', 'F'),
            (         'none', 'M'),
            (        'other', 'F'),
            (        'other', 'M'),
            (   'programmer'

In [53]:
index = pd.MultiIndex.from_tuples(tmp.index)
df_test = tmp.reindex(index)
df_test

age  zip_code
administrator F   36        36
              M   43        43
artist        F   13        13
              M   15        15
doctor        M    7         7
educator      F   26        26
              M   69        69
engineer      F    2         2
              M   65        65
entertainment F    2         2
              M   16        16
executive     F    3         3
              M   29        29
healthcare    F   11        11
              M    5         5
homemaker     F    6         6
              M    1         1
lawyer        F    2         2
              M   10        10
librarian     F   29        29
              M   22        22
marketing     F   10        10
              M   16        16
none          F    4         4
              M    5         5
other         F   36        36
              M   69        69
programmer    F    6         6
              M   60        60
retired       F    1         1
              M   13        13
salesman      F    3         3
              M    9         9
scientist     F    3         3
              M   28        28
student       F   60        60
              M  136       136
technician    F    1         1
              M   26        26
writer        F   19        19
              M   26        26

In [60]:
tmp

age  zip_code
occupation    gender               
administrator F        36        36
              M        43        43
artist        F        13        13
              M        15        15
doctor        M         7         7
educator      F        26        26
              M        69        69
engineer      F         2         2
              M        65        65
entertainment F         2         2
              M        16        16
executive     F         3         3
              M        29        29
healthcare    F        11        11
              M         5         5
homemaker     F         6         6
              M         1         1
lawyer        F         2         2
              M        10        10
librarian     F        29        29
              M        22        22
marketing     F        10        10
              M        16        16
none          F         4         4
              M         5         5
other         F        36        36
              M        69        69
programmer    F         6         6
              M        60        60
retired       F         1         1
              M        13        13
salesman      F         3         3
              M         9         9
scientist     F         3         3
              M        28        28
student       F        60        60
              M       136       136
technician    F         1         1
              M        26        26
writer        F        19        19
              M        26        26

In [63]:
tmp.index

MultiIndex([('administrator', 'F'),
            ('administrator', 'M'),
            (       'artist', 'F'),
            (       'artist', 'M'),
            (       'doctor', 'M'),
            (     'educator', 'F'),
            (     'educator', 'M'),
            (     'engineer', 'F'),
            (     'engineer', 'M'),
            ('entertainment', 'F'),
            ('entertainment', 'M'),
            (    'executive', 'F'),
            (    'executive', 'M'),
            (   'healthcare', 'F'),
            (   'healthcare', 'M'),
            (    'homemaker', 'F'),
            (    'homemaker', 'M'),
            (       'lawyer', 'F'),
            (       'lawyer', 'M'),
            (    'librarian', 'F'),
            (    'librarian', 'M'),
            (    'marketing', 'F'),
            (    'marketing', 'M'),
            (         'none', 'F'),
            (         'none', 'M'),
            (        'other', 'F'),
            (        'other', 'M'),
            (   'programmer'

In [67]:
tmp['age'][:,'M']

occupation
administrator     43
artist            15
doctor             7
educator          69
engineer          65
entertainment     16
executive         29
healthcare         5
homemaker          1
lawyer            10
librarian         22
marketing         16
none               5
other             69
programmer        60
retired           13
salesman           9
scientist         28
student          136
technician        26
writer            26
Name: age, dtype: int64

In [72]:
tmp['age'][:,'F']

occupation
administrator    36
artist           13
educator         26
engineer          2
entertainment     2
executive         3
healthcare       11
homemaker         6
lawyer            2
librarian        29
marketing        10
none              4
other            36
programmer        6
retired           1
salesman          3
scientist         3
student          60
technician        1
writer           19
Name: age, dtype: int64

In [73]:
tmp['age'][:,'M'] + tmp['age'][:,'F']

occupation
administrator     79.0
artist            28.0
doctor             NaN
educator          95.0
engineer          67.0
entertainment     18.0
executive         32.0
healthcare        16.0
homemaker          7.0
lawyer            12.0
librarian         51.0
marketing         26.0
none               9.0
other            105.0
programmer        66.0
retired           14.0
salesman          12.0
scientist         31.0
student          196.0
technician        27.0
writer            45.0
Name: age, dtype: float64

In [74]:
(tmp['age'][:,'M']) / (tmp['age'][:,'M'] + tmp['age'][:,'F']) *100

occupation
administrator    54.430380
artist           53.571429
doctor                 NaN
educator         72.631579
engineer         97.014925
entertainment    88.888889
executive        90.625000
healthcare       31.250000
homemaker        14.285714
lawyer           83.333333
librarian        43.137255
marketing        61.538462
none             55.555556
other            65.714286
programmer       90.909091
retired          92.857143
salesman         75.000000
scientist        90.322581
student          69.387755
technician       96.296296
writer           57.777778
Name: age, dtype: float64

In [75]:
((tmp['age'][:,'M']) / (tmp['age'][:,'M'] + tmp['age'][:,'F']) *100).sort_values()

occupation
homemaker        14.285714
healthcare       31.250000
librarian        43.137255
artist           53.571429
administrator    54.430380
none             55.555556
writer           57.777778
marketing        61.538462
other            65.714286
student          69.387755
educator         72.631579
salesman         75.000000
lawyer           83.333333
entertainment    88.888889
scientist        90.322581
executive        90.625000
programmer       90.909091
retired          92.857143
technician       96.296296
engineer         97.014925
doctor                 NaN
Name: age, dtype: float64

In [77]:
tmp.unstack()

age        zip_code       
gender            F      M        F      M
occupation                                
administrator  36.0   43.0     36.0   43.0
artist         13.0   15.0     13.0   15.0
doctor          NaN    7.0      NaN    7.0
educator       26.0   69.0     26.0   69.0
engineer        2.0   65.0      2.0   65.0
entertainment   2.0   16.0      2.0   16.0
executive       3.0   29.0      3.0   29.0
healthcare     11.0    5.0     11.0    5.0
homemaker       6.0    1.0      6.0    1.0
lawyer          2.0   10.0      2.0   10.0
librarian      29.0   22.0     29.0   22.0
marketing      10.0   16.0     10.0   16.0
none            4.0    5.0      4.0    5.0
other          36.0   69.0     36.0   69.0
programmer      6.0   60.0      6.0   60.0
retired         1.0   13.0      1.0   13.0
salesman        3.0    9.0      3.0    9.0
scientist       3.0   28.0      3.0   28.0
student        60.0  136.0     60.0  136.0
technician      1.0   26.0      1.0   26.0
writer         19.0   26.0     19.0   26.0

In [78]:
tmp.unstack().fillna(0)

age        zip_code       
gender            F      M        F      M
occupation                                
administrator  36.0   43.0     36.0   43.0
artist         13.0   15.0     13.0   15.0
doctor          0.0    7.0      0.0    7.0
educator       26.0   69.0     26.0   69.0
engineer        2.0   65.0      2.0   65.0
entertainment   2.0   16.0      2.0   16.0
executive       3.0   29.0      3.0   29.0
healthcare     11.0    5.0     11.0    5.0
homemaker       6.0    1.0      6.0    1.0
lawyer          2.0   10.0      2.0   10.0
librarian      29.0   22.0     29.0   22.0
marketing      10.0   16.0     10.0   16.0
none            4.0    5.0      4.0    5.0
other          36.0   69.0     36.0   69.0
programmer      6.0   60.0      6.0   60.0
retired         1.0   13.0      1.0   13.0
salesman        3.0    9.0      3.0    9.0
scientist       3.0   28.0      3.0   28.0
student        60.0  136.0     60.0  136.0
technician      1.0   26.0      1.0   26.0
writer         19.0   26.0     19.0   26.0

In [79]:
tmp.unstack().fillna(0)['age']

gender            F      M
occupation                
administrator  36.0   43.0
artist         13.0   15.0
doctor          0.0    7.0
educator       26.0   69.0
engineer        2.0   65.0
entertainment   2.0   16.0
executive       3.0   29.0
healthcare     11.0    5.0
homemaker       6.0    1.0
lawyer          2.0   10.0
librarian      29.0   22.0
marketing      10.0   16.0
none            4.0    5.0
other          36.0   69.0
programmer      6.0   60.0
retired         1.0   13.0
salesman        3.0    9.0
scientist       3.0   28.0
student        60.0  136.0
technician      1.0   26.0
writer         19.0   26.0

In [84]:
F = tmp.unstack().fillna(0)['age'].loc[:,'F']
F

occupation
administrator    36.0
artist           13.0
doctor            0.0
educator         26.0
engineer          2.0
entertainment     2.0
executive         3.0
healthcare       11.0
homemaker         6.0
lawyer            2.0
librarian        29.0
marketing        10.0
none              4.0
other            36.0
programmer        6.0
retired           1.0
salesman          3.0
scientist         3.0
student          60.0
technician        1.0
writer           19.0
Name: F, dtype: float64

In [85]:
M = tmp.unstack().fillna(0)['age'].loc[:,'M']
M

occupation
administrator     43.0
artist            15.0
doctor             7.0
educator          69.0
engineer          65.0
entertainment     16.0
executive         29.0
healthcare         5.0
homemaker          1.0
lawyer            10.0
librarian         22.0
marketing         16.0
none               5.0
other             69.0
programmer        60.0
retired           13.0
salesman           9.0
scientist         28.0
student          136.0
technician        26.0
writer            26.0
Name: M, dtype: float64

In [88]:
F+M

occupation
administrator     79.0
artist            28.0
doctor             7.0
educator          95.0
engineer          67.0
entertainment     18.0
executive         32.0
healthcare        16.0
homemaker          7.0
lawyer            12.0
librarian         51.0
marketing         26.0
none               9.0
other            105.0
programmer        66.0
retired           14.0
salesman          12.0
scientist         31.0
student          196.0
technician        27.0
writer            45.0
dtype: float64

In [89]:
M / (F+M) * 100

occupation
administrator     54.430380
artist            53.571429
doctor           100.000000
educator          72.631579
engineer          97.014925
entertainment     88.888889
executive         90.625000
healthcare        31.250000
homemaker         14.285714
lawyer            83.333333
librarian         43.137255
marketing         61.538462
none              55.555556
other             65.714286
programmer        90.909091
retired           92.857143
salesman          75.000000
scientist         90.322581
student           69.387755
technician        96.296296
writer            57.777778
dtype: float64

In [91]:
(M / (F+M) * 100).sort_values(ascending=False)

occupation
doctor           100.000000
engineer          97.014925
technician        96.296296
retired           92.857143
programmer        90.909091
executive         90.625000
scientist         90.322581
entertainment     88.888889
lawyer            83.333333
salesman          75.000000
educator          72.631579
student           69.387755
other             65.714286
marketing         61.538462
writer            57.777778
none              55.555556
administrator     54.430380
artist            53.571429
librarian         43.137255
healthcare        31.250000
homemaker         14.285714
dtype: float64

### Step 6. 직업별로 나이의 최소값과 최대값을 구해주세요

min  max
occupation             
administrator   21   70
artist          19   48
doctor          28   64
educator        23   63
engineer        22   70
entertainment   15   50
executive       22   69
healthcare      22   62
homemaker       20   50
lawyer          21   53
librarian       23   69
marketing       24   55
none            11   55
other           13   64
programmer      20   63
retired         51   73
salesman        18   66
scientist       23   55
student          7   42
technician      21   55
writer          18   60

In [113]:
users1.groupby('occupation').agg(['min','max'])

age     gender     zip_code       
              min max    min max      min    max
occupation                                      
administrator  21  70      F   M    02154  V1G4L
artist         19  48      F   M    01945  V5A2B
doctor         28  64      M   M    47401  97405
educator       23  63      F   M    01602  M4J2K
engineer       22  70      F   M    00000  T8H1N
entertainment  15  50      F   M    01040  V3N4P
executive      22  69      F   M    00000  L1V3W
healthcare     22  62      F   M    02154  97232
homemaker      20  50      F   M    17331  96349
lawyer         21  53      F   M    06371  90703
librarian      23  69      F   M    01080  99603
marketing      24  55      F   M    01913  97215
none           11  55      F   M    21206  94619
other          13  64      F   M    02110  V0R2M
programmer     20  63      F   M    01331  98121
retired        51  73      F   M    02903  98257
salesman       18  66      F   M    30011  97408
scientist      23  55      F   M    02113  T8H1N
student         7  42      F   M    01002  R3T5K
technician     21  55      F   M    02139  Y1A6B
writer         18  60      F   M    08534  E2A4H

In [93]:
users1.groupby('occupation')['age'].agg(['min','max'])

min  max
occupation             
administrator   21   70
artist          19   48
doctor          28   64
educator        23   63
engineer        22   70
entertainment   15   50
executive       22   69
healthcare      22   62
homemaker       20   50
lawyer          21   53
librarian       23   69
marketing       24   55
none            11   55
other           13   64
programmer      20   63
retired         51   73
salesman        18   66
scientist       23   55
student          7   42
technician      21   55
writer          18   60

### Step 7. 직업별로 성별을 나눠 평균 연령을 구해주세요

(groupby를 직업과 성별 두 개로 해주시면 됩니다)

occupation     gender
administrator  F         40.638889
               M         37.162791
artist         F         30.307692
               M         32.333333
doctor         M         43.571429
educator       F         39.115385
               M         43.101449
engineer       F         29.500000
               M         36.600000
entertainment  F         31.000000
               M         29.000000
executive      F         44.000000
               M         38.172414
healthcare     F         39.818182
               M         45.400000
homemaker      F         34.166667
               M         23.000000
lawyer         F         39.500000
               M         36.200000
librarian      F         40.000000
               M         40.000000
marketing      F         37.200000
               M         37.875000
none           F         36.500000
               M         18.600000
other          F         35.472222
               M         34.028986
programmer     F         32.16666

In [98]:
users1.head()

age gender  occupation zip_code
user_id                                 
1         24      M  technician    85711
2         53      F       other    94043
3         23      M      writer    32067
4         24      M  technician    43537
5         33      F       other    15213

In [99]:
users1.groupby(['occupation', 'gender'])['age'].mean()

occupation     gender
administrator  F         40.638889
               M         37.162791
artist         F         30.307692
               M         32.333333
doctor         M         43.571429
educator       F         39.115385
               M         43.101449
engineer       F         29.500000
               M         36.600000
entertainment  F         31.000000
               M         29.000000
executive      F         44.000000
               M         38.172414
healthcare     F         39.818182
               M         45.400000
homemaker      F         34.166667
               M         23.000000
lawyer         F         39.500000
               M         36.200000
librarian      F         40.000000
               M         40.000000
marketing      F         37.200000
               M         37.875000
none           F         36.500000
               M         18.600000
other          F         35.472222
               M         34.028986
programmer     F         32.16666

### Step 8. 직업별로 각 성별이 몇%를 차지하는지 구해주세요

administrator의 경우: 여성비율 45.56%, 남성비율 54.43%

In [104]:
tmp = users1.groupby(['occupation', 'gender']).count()['age']
tmp

occupation     gender
administrator  F          36
               M          43
artist         F          13
               M          15
doctor         M           7
educator       F          26
               M          69
engineer       F           2
               M          65
entertainment  F           2
               M          16
executive      F           3
               M          29
healthcare     F          11
               M           5
homemaker      F           6
               M           1
lawyer         F           2
               M          10
librarian      F          29
               M          22
marketing      F          10
               M          16
none           F           4
               M           5
other          F          36
               M          69
programmer     F           6
               M          60
retired        F           1
               M          13
salesman       F           3
               M           9
scientist      F     

In [107]:
tmp.unstack(1)

gender            F      M
occupation                
administrator  36.0   43.0
artist         13.0   15.0
doctor          NaN    7.0
educator       26.0   69.0
engineer        2.0   65.0
entertainment   2.0   16.0
executive       3.0   29.0
healthcare     11.0    5.0
homemaker       6.0    1.0
lawyer          2.0   10.0
librarian      29.0   22.0
marketing      10.0   16.0
none            4.0    5.0
other          36.0   69.0
programmer      6.0   60.0
retired         1.0   13.0
salesman        3.0    9.0
scientist       3.0   28.0
student        60.0  136.0
technician      1.0   26.0
writer         19.0   26.0

In [108]:
tmp.unstack(1).fillna(0)

gender            F      M
occupation                
administrator  36.0   43.0
artist         13.0   15.0
doctor          0.0    7.0
educator       26.0   69.0
engineer        2.0   65.0
entertainment   2.0   16.0
executive       3.0   29.0
healthcare     11.0    5.0
homemaker       6.0    1.0
lawyer          2.0   10.0
librarian      29.0   22.0
marketing      10.0   16.0
none            4.0    5.0
other          36.0   69.0
programmer      6.0   60.0
retired         1.0   13.0
salesman        3.0    9.0
scientist       3.0   28.0
student        60.0  136.0
technician      1.0   26.0
writer         19.0   26.0

In [109]:
tmp = tmp.unstack(1).fillna(0)

In [110]:
tmp['F']

occupation
administrator    36.0
artist           13.0
doctor            0.0
educator         26.0
engineer          2.0
entertainment     2.0
executive         3.0
healthcare       11.0
homemaker         6.0
lawyer            2.0
librarian        29.0
marketing        10.0
none              4.0
other            36.0
programmer        6.0
retired           1.0
salesman          3.0
scientist         3.0
student          60.0
technician        1.0
writer           19.0
Name: F, dtype: float64

In [111]:
여성비율 = tmp['F'] / (tmp['F']+tmp['M']) * 100
여성비율

occupation
administrator    45.569620
artist           46.428571
doctor            0.000000
educator         27.368421
engineer          2.985075
entertainment    11.111111
executive         9.375000
healthcare       68.750000
homemaker        85.714286
lawyer           16.666667
librarian        56.862745
marketing        38.461538
none             44.444444
other            34.285714
programmer        9.090909
retired           7.142857
salesman         25.000000
scientist         9.677419
student          30.612245
technician        3.703704
writer           42.222222
dtype: float64

In [112]:
남성비율 = tmp['M'] / (tmp['F']+tmp['M']) * 100
남성비율

occupation
administrator     54.430380
artist            53.571429
doctor           100.000000
educator          72.631579
engineer          97.014925
entertainment     88.888889
executive         90.625000
healthcare        31.250000
homemaker         14.285714
lawyer            83.333333
librarian         43.137255
marketing         61.538462
none              55.555556
other             65.714286
programmer        90.909091
retired           92.857143
salesman          75.000000
scientist         90.322581
student           69.387755
technician        96.296296
writer            57.777778
dtype: float64

In [114]:
users1.head()

user_id  age gender  occupation zip_code
0        1   24      M  technician    85711
1        2   53      F       other    94043
2        3   23      M      writer    32067
3        4   24      M  technician    43537
4        5   33      F       other    15213

In [118]:
tmp = users1.groupby(['occupation', 'gender']).count()['age'].unstack(1)
tmp

gender            F      M
occupation                
administrator  36.0   43.0
artist         13.0   15.0
doctor          NaN    7.0
educator       26.0   69.0
engineer        2.0   65.0
entertainment   2.0   16.0
executive       3.0   29.0
healthcare     11.0    5.0
homemaker       6.0    1.0
lawyer          2.0   10.0
librarian      29.0   22.0
marketing      10.0   16.0
none            4.0    5.0
other          36.0   69.0
programmer      6.0   60.0
retired         1.0   13.0
salesman        3.0    9.0
scientist       3.0   28.0
student        60.0  136.0
technician      1.0   26.0
writer         19.0   26.0

In [120]:
s = tmp.sum(axis=1)
s

occupation
administrator     79.0
artist            28.0
doctor             7.0
educator          95.0
engineer          67.0
entertainment     18.0
executive         32.0
healthcare        16.0
homemaker          7.0
lawyer            12.0
librarian         51.0
marketing         26.0
none               9.0
other            105.0
programmer        66.0
retired           14.0
salesman          12.0
scientist         31.0
student          196.0
technician        27.0
writer            45.0
dtype: float64

In [121]:
tmp['F_1'] = round((tmp['F'] / s * 100),2)
tmp['M_1'] = round((tmp['M'] / s * 100),2)
tmp

gender            F      M    F_1     M_1
occupation                               
administrator  36.0   43.0  45.57   54.43
artist         13.0   15.0  46.43   53.57
doctor          NaN    7.0    NaN  100.00
educator       26.0   69.0  27.37   72.63
engineer        2.0   65.0   2.99   97.01
entertainment   2.0   16.0  11.11   88.89
executive       3.0   29.0   9.38   90.62
healthcare     11.0    5.0  68.75   31.25
homemaker       6.0    1.0  85.71   14.29
lawyer          2.0   10.0  16.67   83.33
librarian      29.0   22.0  56.86   43.14
marketing      10.0   16.0  38.46   61.54
none            4.0    5.0  44.44   55.56
other          36.0   69.0  34.29   65.71
programmer      6.0   60.0   9.09   90.91
retired         1.0   13.0   7.14   92.86
salesman        3.0    9.0  25.00   75.00
scientist       3.0   28.0   9.68   90.32
student        60.0  136.0  30.61   69.39
technician      1.0   26.0   3.70   96.30
writer         19.0   26.0  42.22   57.78